fuel-p2p/src/service.rs

```diff
@@ -332,4 +380,130 @@ mod tests {
+                message_sent = rx_test_end.recv() => {
+                    // we received a signal to end the test
+                    assert_eq!(message_sent, Some(true), "Message not received successfully!");
+                    break;
+                }
+                event_a = node_a.next_event() => {
+                    if let FuelP2PEvent::Behaviour(FuelBehaviourEvent::PeerInfoUpdated(peer_id)) = event_a {
+                        let PeerInfo { peer_addresses, .. } = node_a.swarm.behaviour().get_peer_info(&peer_id).unwrap();
+
+                        // 0. verifies that we've got at least a single peer address to request messsage from
+                        if !peer_addresses.is_empty() {
+                            // 1. Simulating Oneshot channel from the NetworkOrchestrator
+                            let (tx_orchestrator, rx_orchestrator) = oneshot::channel();
+                            assert!(node_a.send_request_msg(None, RequestMessage::RequestBlock, tx_orchestrator).is_ok());
+
+                            let tx_test_end = tx_test_end.clone();
+                            tokio::spawn(async move {
+                                // 4. Simulating NetworkOrchestrator receving a message from Node B
+                                let message_sent = matches!(rx_orchestrator.await, Ok(Ok(_)));
+                                tx_test_end.send(message_sent).await
+                            });
+                        }
+                    }
+
+                },
+                event_b = node_b.next_event() => {
+                    // 2. Node B recieves the RequestMessage from Node A initiated by the NetworkOrchestrator
+                    if let FuelP2PEvent::Behaviour(FuelBehaviourEvent::RequestMessage{ request_id, .. }) = event_b {
+                        let _ = node_b.send_response_msg(request_id, ResponseMessage::ResponseBlock);
+                    }
+                }
+            };
+        }
+    }
+
+    #[tokio::test]
+    async fn req_res_outbound_timeout_works() {
+        let mut p2p_config = build_p2p_config();
+
+        // Node A
+        p2p_config.tcp_port = 4012;
+        // setup request timeout to 0 in order for the Request to fail
+        p2p_config.set_request_timeout = Some(Duration::from_secs(0));
+        let mut node_a = build_fuel_p2p_service(p2p_config).await;
+
+        let node_a_address = match node_a.next_event().await {
+            FuelP2PEvent::NewListenAddr(address) => Some(address),
+            _ => None,
+        };
+
+        // Node B
+        let mut p2p_config = build_p2p_config();
+        p2p_config.tcp_port = 4013;
+        p2p_config.bootstrap_nodes = vec![(node_a.local_peer_id, node_a_address.clone().unwrap())];
+        let mut node_b = build_fuel_p2p_service(p2p_config.clone()).await;
+
+        let (tx_test_end, mut rx_test_end) = tokio::sync::mpsc::channel(1);
+
+        loop {
+            tokio::select! {
+                _ = rx_test_end.recv() => {
```

Comment:
might be easier to follow the steps 1,2,3,.. if we move this match arm down below the event_a handler.